e2e/tests/truffle/contracts/LoomNativeApi.sol

```diff
@@ -0,0 +1,61 @@
+pragma solidity >=0.4.21;
+
+library LoomNativeApi {
+    address constant LoomPrecompilesStartIndex = 0x0000000000000000000000000000000000000020;
+    address constant MapToLoomAddress = 0x0000000000000000000000000000000000000021;
+    address constant MapToAddress = 0x0000000000000000000000000000000000000022;
+
+    function MappedAccount(string memory from, address addr) public view returns (address) {
+        bytes memory fromB = bytes(from);
+        bytes memory input = new bytes(fromB.length + 20);
+        for (uint i = 0; i < 20; i++) {
+            input[i] = byte(uint8(uint(addr) / (2**(8*(19 - i)))));
+        }
+        for (uint j = 0; j < fromB.length; j++) {
+            input[20+j] = fromB[j];
+        }
+        return address(callPFAssembly(MapToLoomAddress, input, 0x14));
+    }
+
+    function MappedAccount(string memory from, address addr, string memory to) public view returns (address) {
+        bytes memory fromB = bytes(from);
+        require(fromB.length < 256, "chain id too long");
+        bytes memory toB = bytes(to);
+        require(toB.length < 256, "chain id too long");
+        bytes memory input = new bytes(fromB.length + toB.length + 21);
+        for (uint i = 0; i < 20; i++) {
+            input[i] = byte(uint8(uint(addr) / (2**(8*(19 - i)))));
+        }
+        input[20] = byte(uint8(fromB.length));
+        for (uint j = 0; j < fromB.length; j++) {
+            input[21+j] = fromB[j];
+        }
+        for (uint k = 0; k < toB.length; k++) {
+            input[21+fromB.length+k] = toB[k];
+        }
+        return address(callPFAssembly(MapToAddress, input, 0x14));
+    }
+
+    function callPFAssembly(address _addr, bytes memory _input, uint256 outSize) view internal returns (bytes20)    {
+        uint256 inSize = _input.length*4+1;
+        uint256 inLenght = _input.length;
+        bytes20 rtv;
+        assembly{
+            let start := add(_input, 0x20)
+            let p := mload(0x40)
+            mstore(p, inSize)
+            if iszero(staticcall(
```

Comment:
Got a link to some docs or example that explains how this works?

Reply:
https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4 is the main guide I used, could not find much else.
Also, https://solidity.readthedocs.io/en/v0.5.12/assembly.html is a reference of the Solidity assembly commands, nothing about precompiles.

Reply:
`call(g, a, v, in, insize, out, outsize)` call contract at address a with input mem[in…(in+insize)) providing g gas and v wei and output area mem[out…(out+outsize)) returning 0 on error (eg. out of gas) and 1 on success


`staticcall(g, a, in, insize, out, outsize)`  identical to call(g, a, 0, in, insize, out, outsize) but do not allow state modifications

Where a = address is the number of our new precompiled functions. Need to take care to get the input and output data setup correctly.